bin/reth/src/utils.rs

```diff
@@ -19,6 +19,20 @@ use reth_provider::insert_canonical_block;
 use std::{collections::BTreeMap, path::Path};
 use tracing::info;
 
+use crate::version;
+
+const VERSION: &str = version::PKG_VERSION;
+const PKG_NAME: &str = version::PKG_NAME;
+const SHA: Option<&str> = version::GIT_COMMIT_HASH_SHORT;
+const OS: &str = std::env::consts::OS;
+const RUSTC: &str = version::RUSTC_VERSION;
+
+/// The version message for the current program, like
+/// `reth/v0.1.0/linux-df71e85/rustc 1.68.2 (9eb3afe9e 2023-03-27)`
+pub fn version_message() -> String {
```

Comment:
I don't think we need at least two variants here:

this one which includes all details, we can use this for error messages etc.. so it can be reported 

for the default extradata we shouldn't use this because this is danger close to 32byte (extrdata requirement)
for this
> reth/v0.1.0/linux-df71e85
is sufficient, don't think we need the rust version and sha

we will use this for the default

Reply:
Okay